rllib/policy/tf_policy.py

```diff
@@ -228,7 +228,7 @@ def _initialize_loss(self, loss, loss_inputs):
         # TODO(sven/ekl): Deprecate support for v1 models.
         if hasattr(self, "model") and isinstance(self.model, ModelV2):
             self._variables = ray.experimental.tf_utils.TensorFlowVariables(
-                [], self._sess, self.variables())
+                self._loss, self._sess, self.variables())
```

Comment:
Does this duplicate some variables in this list? Can we verify this doesn't happen with a test?

Reply:
Yeah, it would produce duplicates. I'm actually not even sure at this point anymore, why we even needed the `self.variables()` method at all. if we can always get everything from traversing the graph (like in ModelV1). I'll make sure, the returned vars are unique. ...

Reply:
Yeah the benefit is somewhat unclear. In principle, having the model declare the variables is more explicit which makes sense. But if we have to do this autodetection of optimizer variables after the fact that kind of breaks the abstraction.

Maybe we can have this only detect optimizer variables, and either (1) raise an error if some variables are not included, or (2) ignore / warn of non-optimizer variables?